src/Trie.js

```diff
@@ -7,33 +7,20 @@ const getClosestNode = require('./getClosestNode')
 /**
   Trie class
   @name Trie
-  @example const store = new Trie()
+  @example const store = new Trie(...args)
 */
-class Trie {
+class Trie extends Map {
   rootSocket: Object
-  cache: Map<string, any>
-  size: number
 
-  constructor () {
+  constructor (args?: any) {
+    super(args)
+    const self = this
     this.rootSocket = {}
-    this.cache = new Map()
-    this.size = 0 // TODO
-  }
-
-  entries () {
-    // TODO
-  }
-
-  has () {
-    // TODO
-  }
-
-  keys () {
-    // TODO
-  }
-
-  values () {
-    // TODO
+    if (args) {
+      args.forEach((value, key, table) => {
+        self.set(key, value)
+      })
+    }
   }
 
   [Symbol.iterator] () {
```

Comment:
Could you explain why do you ever need this? Isn't it something that you've already extended from the Map?

Reply:
it's becouse we have our implementation of `set` for our structure, and works different

Reply:
```js
[Symbol.iterator] () {
```

I mean this snippet

Reply:
ahh. yes. i've already have it